### PR TITLE
Fix sibling block inserter displaying at end of block list.

### LINKED ISSUE #28907: Double inserter button showing at the end of some block lists
## Description
In some block lists (like the navigation block) the trailing appender shows as well as a sibling inserter.

This was introduced by #28418. The solution is probably that the insertion point can be displayed after the last block when hovering over blocks in the block library or quick inserter menu. It shouldn't display when there's a trailing appender already.

## Step-by-step reproduction instructions
1. Add a navigation block
2. Add some menu items
3. Hover around the trailing edge of the last menu item

## Expected behaviour
Only one inserter

## Actual behaviour
It's very insertery

## Screenshots or screen recording (optional)
https://user-images.githubusercontent.com/107534/106244522-af4ea680-6213-11eb-8525-9991aed82213.mp4

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #28907

## How has this been tested?
1. Add a Navigation Block
2. Start empty and add a couple of links
3. Hover in the area after the last link
4. The inserter should not be displayed
5. Hover between the links
6. The inserter should be displayed
7. Select the last link
8. Click the 'Add block' button from the editor's top toolbar
9. Hover over a block in the block library
10. The insertion point (blue line) should be displayed at the end of the navigation block.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-list/insertion-point.js [hunk 1]
```diff
@@ -200,6 +200,18 @@ function InsertionPointPopover( {
...
+	// insertion point). At the end of the block list the trailing appender
+	// should serve the purpose of inserting blocks.
+	const showInsertionPointInserter =
+		! isHidden && nextElement && ( isInserterShown || isInserterForced );
+
+	// Show the indicator if the insertion point inserter is visible, or if
+	// the `showInsertionPoint` state is `true`. The latter is generally true
+	// when hovering blocks for insertion in the block library.
+	const showInsertionPointIndicator =
+		! isHidden && ( showInsertionPointInserter || showInsertionPoint );
```
Comment (talldan) [inline packages/block-editor/src/components/block-list/insertion-point.js]: `! isHidden` logic is duplicated, but I think this is one of those situations where avoiding that would make the code much uglier/harder to follow.

Comment (ellatrix) [reply packages/block-editor/src/components/block-list/insertion-point.js]: `showInsertionPointInserter || ( ! isHidden && showInsertionPoint )`?

Comment (talldan) [reply packages/block-editor/src/components/block-list/insertion-point.js]: Yeah, that's better 😄

**Comment by talldan** [comment]
I'm not completely sure why this suddenly caused a consistent test failure, but the test was selecting a template part by clicking, and was failing because it clicked the inserter between two columns in the template part.

Testing manually, the same thing happens in `trunk` and this branch, the inserter between the two columns can be clicked.

It's actually pretty difficult to select the template part by clicking the block itself, so I've updated the test to use List View to select the block.

Edit: same issue with the document settings test as well.

File: packages/block-editor/src/components/block-list/insertion-point.js [hunk 2]
```diff
@@ -200,6 +200,18 @@ function InsertionPointPopover( {
 		}
 	}
 
+	// Only show the inserter when there's a `nextElement` (a block after the
+	// insertion point). At the end of the block list the trailing appender
+	// should serve the purpose of inserting blocks.
```
Comment (ellatrix) [inline packages/block-editor/src/components/block-list/insertion-point.js:205]: What if there's no appender available?

Comment (talldan) [reply packages/block-editor/src/components/block-list/insertion-point.js:205]: I don't think that's an issue, from what I understand this inserter was never intended to show at the end of a block list. Its appearance there is a bug from https://github.com/WordPress/gutenberg/pull/28418.

**Review by ellatrix — APPROVED** [review]

**Comment by talldan** [comment]
Thanks for reviewing @ellatrix.

**Comment by youknowriad** [comment]
Thanks for fixing this :) I appreciate it.

**Pull request merged**